src/main/java/com/ss/training/utopia/agent/service/AgentService.java

```diff
@@ -1,29 +1,160 @@
+		}
+		if (flight.getSeatsAvailable() <= 0)
+            return "Flight Full";
+
+        try {
+            Charge charge = stripePurchase(booking);
+            booking.setStripeId(charge.getId());
+            bookingDAO.save(booking);
+          } catch (CardException e) {
+            return "Card Declined";
+          } catch (Exception e) {
+            return "Internal Server Error";
+            } catch (Throwable t) {
+                return "Internal Server Error";
+            }
+
+        return "Charge Created";
     }
 
-}
\ No newline at end of file
+
+    /**
+     * 
+     * @param booking
+     * @return
+     * @throws AuthenticationException
+     * @throws InvalidRequestException
+     * @throws APIConnectionException
+     * @throws CardException
+     * @throws APIException
+     */
+    public Charge stripePurchase(Booking booking) throws AuthenticationException, InvalidRequestException,
+    APIConnectionException, CardException, APIException {
+        
+        Integer flightPrice = (int) (flightDAO.findByFlightId(booking.getFlightId()).getPrice() * 100);
+
+        Stripe.apiKey = stripeKey;
+
+        Map<String, Object> params = new HashMap<>();
+        params.put("amount", flightPrice);
+        params.put("currency", "usd");
+        params.put("source", booking.getStripeId());
+
+        Charge charge = Charge.create(params);
+        return charge;
+        }
+
+    /**
+     * 
+     * @param bookerId
+     * @return
+     */
+    public List<Booking> readAgentBookings(Long bookerId) {
+        try {
+            List<Booking> bookings = bookingDAO.findByBookerId(bookerId);
+            return bookings;
+        } catch (Throwable t) {
+            return null;
+        }
+    }
+
+    /**
+     * 
+     * @param booking
+     * @return
+     */
+    public String cancelBooking(Booking booking) {
+
+        try {
+            stripeRefund(booking);
+            booking.setActive(false);
+            bookingDAO.save(booking);
+        } catch (InvalidRequestException e) {
+            return "Already Refunded";
+        } catch (Exception e) {
+            return "Internal Server Error";
+        } catch (Throwable t) {
+            return "Internal Server Error";
+        }
+
+        return "Refund Processed";
+    }
+
+
+    /**
+     * 
+     * @param booking
+     * @return
+     * @throws AuthenticationException
+     * @throws InvalidRequestException
+     * @throws APIConnectionException
+     * @throws CardException
+     * @throws APIException
+     */
+    public Refund stripeRefund(Booking booking) throws AuthenticationException, InvalidRequestException,
+            APIConnectionException, CardException, APIException {
+        
+        Stripe.apiKey = stripeKey;
+
+		Map<String, Object> params = new HashMap<>();
+            params.put(
+            "charge",
+            booking.getStripeId()
+            );
+
+        Refund refund = Refund.create(params);
+        return refund;
```

Comment:
It seems like this function doesn't need to return anything (unless having a return value makes it more testable).